### ISSUE TITLE
`display: none` does not work as expected with `position: absolute`

### ISSUE THREAD

**Issue by rozele** [microsoft/react-native-windows#7289]
## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Add an example to [ViewExample.windows.js](https://github.com/microsoft/react-native-windows/blob/master/packages/%40react-native-windows/tester/src/js/examples/View/ViewExample.windows.js) in RNTester:
```js
  {
    title: 'display: none and position: absolute',
    render: function(): React.Node {
      return (
        <>
          <View
            style={{
              height: 100,
              width: 100,
              justifyContent: 'center',
              alignItems: 'center',
              backgroundColor: 'red',
              position: 'absolute',
              display: 'none',
              top: 0,
            }}>
            <Text>This should be hidden.</Text>
          </View>
        </>
      );
    }
  }
```
 
2. Run the example in RNTester
3. Observe that the view is still displayed.

## Expected Results
The `display: none` prop value should result in the view being hidden.

## Snack, code example, screenshot, or link to a repository:
https://github.com/rozele/react-native-windows/commit/e6115418824f4c2509cfa1588b7304d84d82eb10

![image](https://user-images.githubusercontent.com/1106239/110151026-f51c0300-7dad-11eb-81fb-492dfdb16711.png)

**Comment by NickGerleman** [comment]
FYI that "display: none" also doesn't propagate to XAML visibility (for focus, UIA, etc).  https://github.com/microsoft/react-native-windows/issues/6355

**Comment by rozele** [comment]
@NickGerleman - this is what we had in the old C# days:
https://github.com/microsoft/react-native-windows/blob/0.57-stable/ReactWindows/ReactNative/UIManager/BaseViewManager.cs#L127

**Comment by chrisglein** [comment]
@rozele is it expected that this is fixed by your above yoga PR?

We also have an issue where collapsed visibility isn't being propagated to XAML. (maybe part of fix for #6355)

We've got a couple "display:none" issues floating around here, trying to making sure we're tracking it all.

**Comment by rozele** [comment]
This can be closed the next time https://github.com/microsoft/react-native-windows/blob/master/vnext/ReactCommon/Yoga.cpp is updated from react-native.
